File: BACK/src/infrastructure/mappers/business.mapper.ts
import { BusinessEntity } from '../../domain/entities/business.entity';
import { IBusiness } from '../datasources/mongo/models/business.model';

export class BusinessMapper {
  static fromMongo(document: IBusiness): BusinessEntity {
    return {
      id: document._id.toString(),
      name: document.name,
      logoUrl: document.logoUrl,
      primaryColor: document.primaryColor,
      secondaryColor: document.secondaryColor,
      slug: document.slug,
      subscriptionPlan: document.subscriptionPlan as any,
      subscriptionStartDate: document.subscriptionStartDate,
      subscriptionEndDate: document.subscriptionEndDate,
      storageUsed: document.storageUsed,
      qrCodeUrl: document.qrCodeUrl,
      isActive: document.isActive,
      createdAt: document.createdAt,
      updatedAt: document.updatedAt,
    };
  }
}

File: AGENDLY_APP/lib/features/auth/infrastructure/mappers/business_mapper.dart
import 'package:agendly_app/features/auth/domain/entities/business.dart';

class BusinessMapper {
  static Business fromJson(Map<String, dynamic> json) {
    return Business(
      id: json['id'] ?? json['_id'] ?? '',
      name: json['name'] ?? '',
      slug: json['slug'] ?? '',
      logoUrl: json['logoUrl'],
      primaryColor: json['primaryColor'] ?? '#0284C7',
      secondaryColor: json['secondaryColor'] ?? '#0EA5E9',
    );
  }

  static Map<String, dynamic> toJson(Business business) {
    return {
      'id': business.id,
      'name': business.name,
      'slug': business.slug,
      'logoUrl': business.logoUrl,
      'primaryColor': business.primaryColor,
      'secondaryColor': business.secondaryColor,
    };
  }
}
